packages/lib/chains/chains-solana/src/index.ts

```diff
@@ -0,0 +1,914 @@
+                p_hash: new Uint8Array(pHash),
+                to: new Uint8Array(base58.decode(to)),
+                n_hash: new Uint8Array(nHash),
+                amount: new BN(amount),
+            },
+            renvmmsg,
+        );
+
+        const mintLogAccountId = await PublicKey.findProgramAddress(
+            [keccak256(renvmmsg)],
+            program,
+        );
+
+        const mintData = await this.provider.connection.getAccountInfo(
+            mintLogAccountId[0],
+        );
+
+        if (!mintData) return undefined;
+        this._logger.debug("found mint:", mintData);
+
+        const mintLogData = MintLogLayout.decode(mintData.data);
+        if (!mintLogData.is_initialized) return undefined;
+
+        const mintSigs = await this.provider.connection.getConfirmedSignaturesForAddress2(
+            mintLogAccountId[0],
+        );
+        return mintSigs[0].signature;
+    };
+
+    /**
+     * Fetch the mint and burn fees for an asset.
+     */
+    getFees(_asset: string) {
+        // TODO: add getFees RPC endpoint; use RPC to provide fees
+        return { burn: 15, mint: 15 };
+    }
+
+    /**
+     * Fetch the addresses' balance of the asset's representation on the chain.
+     */
+    async getBalance(asset: string, address: SolAddress) {
+        const tokenMintId = await this.getSPLTokenPubkey(asset);
+
+        const source = await getAssociatedTokenAddress(
+            new PublicKey(address),
+            tokenMintId,
+        );
+        return new BigNumber(
+            (
+                await this.provider.connection.getTokenAccountBalance(source)
+            ).value.amount,
+        );
+    }
+
+    getMintParams = async (asset: string) => {
+        await this.waitForInitialization();
+        if (!this.renNetworkDetails || !this.provider) {
+            throw new Error(
+                `Solana must be initialized before calling 'getContractCalls'.`,
+            );
+        }
+
+        const contract = this.resolveTokenGatewayContract(asset);
+        const program = new PublicKey(contract);
+
+        const s_hash = keccak256(Buffer.from(asset + "/toSolana"));
+
+        const tokenMintId = await PublicKey.findProgramAddress(
+            [s_hash],
+            program,
+        );
+
+        const destination = await getAssociatedTokenAddress(
+            this.provider.wallet.publicKey,
+            tokenMintId[0],
+        );
+        const tokenAccount = await this.provider.connection.getAccountInfo(
+            destination,
+        );
+        if (
+            !this._creatingTokenAccount &&
+            (!tokenAccount || !tokenAccount.data)
+        ) {
+            this._creatingTokenAccount = true;
+            const createTxInstruction = await createAssociatedTokenAccount(
+                this.provider.wallet.publicKey,
+                this.provider.wallet.publicKey,
+                tokenMintId[0],
+            );
+            const createTx = new Transaction();
```

Comment:
Would it make sense to have another step in the RenJS flow for transactions that have to happen before the mint is initiated?

Otherwise we could make `getMintParams` return a PromiEvent so that integrators can capture information about this transaction.

Reply:
This was another question; how many other chains will require a "token initialization" step? If Solana is the only one, we can reasonably skip this transaction and leave a note to users that they are required to initialize the account on their end first.

There is an alternative method where we can include this transaction in the mint if it's not present; but this will require a modification to the Solana program that Rohit hasn't had time for yet.

Either way, I agree that we should change this, and I would rather not ammend the flow to include more events (as they will need to flow down to all UIs/libs and add more edge cases)

Reply:
It works quite well for now, since solana transactions are cheap and fast, and I would think that removing it would be confusing to integrators.

We can probably leave it as it is for now until a better solution comes up.